rct229/rule_engine/rule_base.py

```diff
@@ -113,26 +114,33 @@ def evaluate(self, rmrs, data=None):
                     # Get calculated values; these can be used by
                     # manual_check_required() or rule_check() and will
                     # be included in the output
-                    calc_vals = self.get_calc_vals(context, data)
-                    if calc_vals is not None:
-                        outcome["calc_vals"] = calc_vals
-
-                    # Determine if manual check is required
-                    if self.manual_check_required(context, calc_vals, data):
-                        outcome["result"] = "MANUAL_CHECK_REQUIRED"
-                    else:
-                        # Evaluate the actual rule check
-                        result = self.rule_check(context, calc_vals, data)
-                        if isinstance(result, list):
-                            # The result is a list of outcomes
-                            outcome["result"] = result
-                        # Assume result type is bool
-                        elif result:
-                            outcome["result"] = "PASSED"
+                    try:
```

Comment:
I think this try should be moved up to include the applicability check.
The except clauses should then be moved down to match.

Reply:
> I'm still not convinced of the need for an exception status, but it isn't hearting anything.

I can comment it out for now.

Reply:
Make sense, I will move the try and catch to include applicability check